crates/pathfinder/src/rpc/v02/method/estimate_fee.rs

```diff
@@ -0,0 +1,418 @@
+    EstimateFeeError: BlockNotFound,
+    ContractNotFound,
+    ContractError,
+    InvalidMessageSelector,
+    InvalidCallData
+);
+
+impl From<crate::cairo::ext_py::CallFailure> for EstimateFeeError {
+    fn from(c: crate::cairo::ext_py::CallFailure) -> Self {
+        use crate::cairo::ext_py::CallFailure::*;
+        match c {
+            NoSuchBlock => Self::BlockNotFound,
+            NoSuchContract => Self::ContractNotFound,
+            InvalidEntryPoint => Self::InvalidMessageSelector,
+            ExecutionFailed(e) => Self::Internal(anyhow::anyhow!("Internal error: {}", e)),
+            // Intentionally hide the message under Internal
+            Internal(_) | Shutdown => Self::Internal(anyhow::anyhow!("Internal error")),
+        }
+    }
+}
+
+impl TryInto<crate::rpc::v01::types::request::Call> for BroadcastedTransaction {
+    type Error = EstimateFeeError;
+
+    fn try_into(self) -> Result<crate::rpc::v01::types::request::Call, Self::Error> {
+        match self {
+            BroadcastedTransaction::Declare(_) | BroadcastedTransaction::Deploy(_) => {
+                Err(EstimateFeeError::Internal(anyhow::anyhow!(
+                    "Internal error: Only invoke version 0 transactions are supported."
+                )))
+            }
+            BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V0(tx)) => {
+                Ok(crate::rpc::v01::types::request::Call {
+                    contract_address: tx.contract_address,
+                    calldata: tx.calldata,
+                    entry_point_selector: Some(tx.entry_point_selector),
+                    signature: tx
+                        .signature
+                        .into_iter()
+                        .map(|x| crate::core::CallSignatureElem(x.0))
+                        .collect(),
+                    max_fee: tx.max_fee,
+                    version: TransactionVersion::ZERO,
+                    nonce: tx.nonce,
+                })
+            }
+            BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V1(tx)) => {
+                Ok(crate::rpc::v01::types::request::Call {
+                    contract_address: tx.sender_address,
+                    calldata: tx.calldata,
+                    entry_point_selector: None,
+                    signature: tx
+                        .signature
+                        .into_iter()
+                        .map(|x| crate::core::CallSignatureElem(x.0))
+                        .collect(),
+                    max_fee: tx.max_fee,
+                    version: TransactionVersion::ZERO,
+                    nonce: tx.nonce,
+                })
+            }
+        }
+    }
+}
+
+pub async fn estimate_fee(
+    context: RpcContext,
+    input: EstimateFeeInput,
+) -> Result<FeeEstimate, EstimateFeeError> {
+    let handle = context
+        .call_handle
+        .as_ref()
+        .ok_or_else(|| EstimateFeeError::Internal(anyhow::anyhow!("Unsupported configuration")))?;
```

Comment:
This should work as well
```suggestion
        .ok_or_else(|| anyhow::anyhow!("Unsupported configuration"))?;
```

Reply:
Or even just
```suggestion
        .context("Unsupported configuration")?;
```